File: support_tools/swigwin-3.0.12/Examples/test-suite/go/smart_pointer_overload_runme.go
package main

import . "./smart_pointer_overload"

func main() {
	f := NewFoo()
	b := NewBar(f)

	if f.Test(3) != 1 {
		panic(0)
	}
	if f.Test(3.5) != 2 {
		panic(0)
	}
	if f.Test("hello") != 3 {
		panic(0)
	}

	if b.Test(3) != 1 {
		panic(0)
	}
	if b.Test(3.5) != 2 {
		panic(0)
	}
	if b.Test("hello") != 3 {
		panic(0)
	}
}

File: support_tools/swigwin-3.0.12/Examples/test-suite/lua/smart_pointer_overload_runme.lua
require("import")	-- the import fn
import("smart_pointer_overload")	-- import code
for k,v in pairs(smart_pointer_overload) do _G[k]=v end -- move to global

f = Foo()
b = Bar(f)

assert(f:test(3) == 1)
--assert(f:test(3.5) == 2)	-- won't work due to being unable to overloads
assert(f:test("hello") == 3)

assert(b:test(3) == 1)
--assert(b:test(3.5) == 2)	-- won't work due to being unable to overloads
assert(b:test("hello") == 3)
